pyproject.toml

```diff
@@ -40,11 +40,11 @@ pydantic = {version = "^2.8.2", optional = true }
 fastapi = {version = ">=0.111.1", optional = true }
 pyyaml = {version = "^6.0.1", optional = true }
 sqlalchemy = {version = "^2.0.36", optional = true }
-psycopg2 = {version = "^2.9.9", optional = true }
+psycopg2-binary = {version = "^2.9.9", optional = true }
 alembic = {version = "^1.13.3", optional = true }
 
 [tool.poetry.extras]
-server = ["pydantic", "fastapi", "pyyaml", "sqlalchemy", "psycopg2", "alembic"]
+server = ["pydantic", "fastapi", "pyyaml", "sqlalchemy", "psycopg2-binary", "alembic"]
```

Comment:
Since you are already using extras, maybe you could define another one, called let's say `server-source`, that would include `psycopg2` instead of `psycopg2-binary`, and you could use that one downstream:
```specfile
%generate_buildrequires
%pyproject_buildrequires -x server-source
```

Reply:
Thanks a lot @nforro I think this is another great idea, better than having a `sed` expression downstream. I try it now!